### PR TITLE
Add 1ES CI for DTFx.AS v2

### PR DESCRIPTION
**Pull request by davidmrdavid**
The primary goal of this PR is to add a public 1ES ADO testing CI for DTFx.AS v2. This pipeline's results (and a messy development history) can be found here: https://dev.azure.com/azfunc/public/_build?definitionId=881&_a=summary

While migrating our CI pipeline, I encountered previously stable tests becoming flaky, as well as the fact that some tests were previously not running (and were failing once enabled) plus other 1ES-specific idiosyncrasies. This PR aims to resolve them all. 

To help with reviews, I'll add comments throughout the diff to clarify why some part of the code had to be changed. I feel comfortable about most changes except for one: the addition of  `System.Runtime.CompilerServices.Unsafe  v6.0.0` as an explicit dependency in DTFx.Core.

Without this change, some DTFx.Core would fail with the following exception trace:

```
Could not load file or assembly 'System.Runtime.CompilerServices.Unsafe, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. Could not find or load a specific file. (0x80131621)
File name: 'System.Runtime.CompilerServices.Unsafe, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
at System.Diagnostics.ActivityTraceId.SetToRandomBytes(Span1 outBytes)    at System.Diagnostics.ActivitySpanId.CreateRandom()    at DurableTask.Core.TaskOrchestrationDispatcher.ProcessScheduleTaskDecision(ScheduleTaskOrchestratorAction scheduleTaskOrchestratorAction, OrchestrationRuntimeState runtimeState, Boolean includeParameters, Activity parentTraceActivity) in /_/src/DurableTask.Core/TaskOrchestrationDispatcher.cs:line 1016    at DurableTask.Core.TaskOrchestrationDispatcher.OnProcessWorkItemAsync(TaskOrchestrationWorkItem workItem) in /_/src/DurableTask.Core/TaskOrchestrationDispatcher.cs:line 412    at DurableTask.Core.TaskOrchestrationDispatcher.OnProcessWorkItemAsync(TaskOrchestrationWorkItem workItem) in /_/src/DurableTask.Core/TaskOrchestrationDispatcher.cs:line 598    at DurableTask.Core.TaskOrchestrationDispatcher.OnProcessWorkItemSessionAsync(TaskOrchestrationWorkItem workItem) in /_/src/DurableTask.Core/TaskOrchestrationDispatcher.cs:line 211    at DurableTask.Core.WorkItemDispatcher1.ProcessWorkItemAsync(WorkItemDispatcherContext context, Object workItemObj) in /_/src/DurableTask.Core/WorkItemDispatcher.cs:line 374
```

This exception occurs in the Distributed Tracing path, as you can see by the mention of `System.Diagnostics.ActivitySpanId` in the stack, which is a type we get from our `System.Diagnostics.DiagnosticSource` dependency.

I tried researching this, and found this [GH thread](https://github.com/dotnet/runtime/issues/67196#issuecomment-1081657399) with the following excerpt:

> Unsafe package isn't referenced anymore on netstandard2.0 or net462 which means that consumers of this package now get the transitive Unsafe dependency of System.Memory which is super old (4.5.3 vs 6.0.0).

Since we recently changed DTFx.Core to use only netstandard2.0 (as per https://github.com/Azure/durabletask/pull/1125/files), it's possible this is the reason why this error is only showing up now. Unfortunately, I'm unable to reproduce this locally, but this reproduces pretty consistently on the 1ES CI. 

**As a relevant aside:** Durable Functions v2 has been incompatible with Azure Functions v1 for a while now. We never got to the bottom of that investigation, but I have a rough recollection that the reason for it was an exception _like this_ being thrown in Functions V1. Perhaps that's worth revisiting.

In any case - I need some .NET expertise to help me validate what's the problem here and if the right solution is indeed to add the explicit dependency to `CompilerServices.Unsafe v6.x`.

Other than that, I'll annotate the PR below. Thanks

_See the CI passing for the latest commit here: https://dev.azure.com/azfunc/public/_build/results?buildId=172877&view=results_

### CODE REVIEW

File: src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj [hunk 1]
```diff
@@ -40,7 +40,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Azure.Core" Version="1.40.0" />
+    <PackageReference Include="Azure.Core" Version="1.41.0" />
```
Comment (davidmrdavid) [inline src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj:43]: 1ES reports Azure.Core 1.40.0 as vulnerable

File: src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs [hunk 2]
```diff
@@ -513,7 +513,7 @@ public override async IAsyncEnumerable<OrchestrationState> GetStateAsync(IEnumer
                 yield break;
             }
 
-            IEnumerable<Task<OrchestrationState>> instanceQueries = instanceIds.Select(instance => this.GetStateAsync(instance, allExecutions: true, fetchInput: false, cancellationToken).SingleAsync().AsTask());
+            IEnumerable<Task<OrchestrationState>> instanceQueries = instanceIds.Select(instance => this.GetStateAsync(instance, allExecutions: true, fetchInput: false, cancellationToken).SingleOrDefaultAsync().AsTask());
```
Comment (davidmrdavid) [inline src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs:516]: There used to be a bug here (which the new timing differences in 1ES revealed). It's possible for `GetStateAsync` to return an empty collection (if there's no state in the instance table for a given ID yet). If that's the case, then `SingleAsync` will throw an error of the type `Sequence contains no elements`. 

By using `SingleOrDefaultAsync`, we make DTFx.Core tolerate this case and return `null` if the collection is empty. That allows us to more gracefully exit this method in line ~522 below.

File: test/DurableTask.AzureStorage.Tests/AzureStorageScenarioTests.cs [hunk 3]
```diff
@@ -2252,6 +2252,7 @@ public async Task ScheduledStart_Activity_GetStatus_Returns_ScheduledStart(bool
         /// End-to-end test which validates a simple orchestrator function that calls an activity function
         /// and checks the OpenTelemetry trace information
         /// </summary>
+        [TestCategory("DisabledInCI")]
```
Comment (davidmrdavid) [inline test/DurableTask.AzureStorage.Tests/AzureStorageScenarioTests.cs:2255]: these disabled tests have **not** been running for a long time, and they're failing in the main branch as well. They weren't running because they're conditionally compiled not to run in NET462, and our previous testing task in ADO was configured to run tests found in the `net462` dll.

Now, the test task does discover these tests, but I've disabled them as they aren't passing. @bachuv can you please add this as a To-Fix before Dist. Tracing GA?

File: test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj [hunk 4]
```diff
@@ -5,28 +5,31 @@
...
   <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.2.0" />
     <PackageReference Include="Microsoft.WindowsAzure.ConfigurationManager" version="3.2.1" />
     <PackageReference Include="WindowsAzure.Storage" version="8.7.0" />
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' != 'net462'">
     <PackageReference Include="Azure.Monitor.OpenTelemetry.Exporter" Version="1.0.0-beta.3" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.6" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (davidmrdavid) [inline test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj:18]: newtonsoft.json is inherited by the DTFx.Core project

File: test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj [hunk 5]
```diff
@@ -5,28 +5,31 @@
...
     <PackageReference Include="System.Configuration.ConfigurationManager" Version="4.5.0" />
     <PackageReference Include="WindowsAzure.Storage" version="9.3.3" />
   </ItemGroup>
 
   <!-- Common package dependencies -->
   <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="2.2.0" />
```
Comment (davidmrdavid) [inline test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj:22]: I added this `Logging.Console` dependency because I've configured the tests to emit traces to STDOUT, which makes debugging failed ADO tests much easier.  I decided to unify the dependencies to version 2.2.0 for simplicy sake and to avoid conditional compilation.

File: test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj [hunk 6]
```diff
@@ -48,9 +51,6 @@
     <None Update="large.jpeg">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
     </None>
-    <None Update="testhost.dll.config" Condition="'$(TargetFramework)' == 'netcoreapp2.1'">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </None>
```
Comment (davidmrdavid) [inline test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj:53]: we don't have a netcoreapp2.1 TFM

File: test/DurableTask.AzureStorage.Tests/TestHelpers.cs [hunk 7]
```diff
@@ -40,9 +41,9 @@ static class TestHelpers
                 TaskHubName = GetTestTaskHubName(),
 
                 // Setting up a logger factory to enable the new DurableTask.Core logs
-                // TODO: Add a logger provider so we can collect these logs in memory.
-                LoggerFactory = new LoggerFactory(),
+                LoggerFactory = new LoggerFactory().AddConsole(LogLevel.Trace),
             };
+#pragma warning restore CS0618 // Type or member is obsolete
```
Comment (davidmrdavid) [inline test/DurableTask.AzureStorage.Tests/TestHelpers.cs:46]: as mentioned, I've added logging to STDOUT on our tests to facilitate debugging. The current logging library we use is pretty old, which is why we have to disable the warning here. I tried using a newer library, but then had issues with the NET462 TFM. For speed, I decided to use the old library instead, which works.

File: test/DurableTask.Core.Tests/DurableTask.Core.Tests.csproj [hunk 8]
```diff
@@ -5,19 +5,13 @@
     <TargetFrameworks>netcoreapp3.1;net462</TargetFrameworks>
   </PropertyGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
-    <PackageReference Include="EnterpriseLibrary.SemanticLogging.EventSourceAnalyzer" Version="2.0.1406.1" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
-    <PackageReference Include="MSTest.TestAdapter" Version="1.4.0" />
-    <PackageReference Include="MSTest.TestFramework" Version="1.4.0" />
-  </ItemGroup>
-
```
Comment (davidmrdavid) [inline test/DurableTask.Core.Tests/DurableTask.Core.Tests.csproj:14]: there was no reason for separate dependencies anymore

File: test/DurableTask.Core.Tests/RetryInterceptorTests.cs [hunk 9]
```diff
@@ -31,7 +31,7 @@ public async Task Invoke_WithFailingRetryCall_ShouldThrowCorrectException()
             await Assert.ThrowsExceptionAsync<IOException>(Invoke, "Interceptor should throw the original exception after exceeding max retry attempts.");
         }
 
-        [TestMethod]
+        [DataTestMethod]
```
Comment (davidmrdavid) [inline test/DurableTask.Core.Tests/RetryInterceptorTests.cs:34]: this was throwing a warning. `TestMethod`s cannot have `DataRows`, for that you need `DataTestMethod` instead. The VSTesting task is forgiving enough here, but I changed this to remove the warning

File: test/DurableTask.Emulator.Tests/DurableTask.Emulator.Tests.csproj [hunk 10]
```diff
@@ -7,8 +7,8 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.7.1" />
-    <PackageReference Include="MSTest.TestAdapter" Version="3.1.1" />
-    <PackageReference Include="MSTest.TestFramework" Version="3.1.1" />
+    <PackageReference Include="MSTest.TestAdapter" Version="1.4.0" />
+    <PackageReference Include="MSTest.TestFramework" Version="1.4.0" />
```
Comment (davidmrdavid) [inline test/DurableTask.Emulator.Tests/DurableTask.Emulator.Tests.csproj:11]: We have an inconsistent use of MSTest, some use version 1.4.0 and others use versoin 3.1.1. The use of different frameworks led to warnings. I've downgraded to the common denominator, as I encountered issues upgrading them all to 3.1.1 (can't remember which unfortunately).

File: test/DurableTask.Test.Orchestrations/SimpleOrchestrations.cs [hunk 11]
```diff
@@ -409,6 +409,11 @@ public async override Task<string> RunTask(OrchestrationContext context, bool us
                 responderOrchestration = Task.FromResult("Herkimer is done");
             }
 
+            // before sending the event, wait a few seconds to ensure the sub-orchestrator exists
+            // otherwise, we risk a race condition where the event is dicarded because the instances table
+            // does not yet have the sub-orchestrator instance in it.
+            await context.CreateTimer<object>(context.CurrentUtcDateTime.AddSeconds(10), state: null);
+
```
Comment (davidmrdavid) [inline test/DurableTask.Test.Orchestrations/SimpleOrchestrations.cs:416]: this test suddenly became flaky. So added this to make it less flakey

File: eng/ci/public-build.yml [hunk 12]
```diff
@@ -67,7 +67,8 @@ extends:
         # Run tests
         - template: /eng/templates/test.yml@self
           parameters:
-            testAssembly: '**\bin\**\netcoreapp3.1\DurableTask.Core.Tests.dll'
+            # TODO: for some reason we're unable to run the net462 tests in the CI
+            testAssembly: '**\bin\Debug\netcoreapp3.1\DurableTask.Core.Tests.dll'
```
Comment (davidmrdavid) [inline eng/ci/public-build.yml:71]: In theory, I want to have the pattern `\**\bin\**\DurableTask.Core.Tests.dll` here. However, when I input that, 1ES complains that there's not tests to be found in the `net462` dll. This makes no sense to me since I can run those tests locally. If I may, I'd like to push for this to remain a "ToDo"

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs [hunk 13]
```diff
@@ -38,8 +40,14 @@ public class DispatcherMiddlewareTests
         [TestInitialize]
         public async Task Initialize()
         {
+            // configure logging so traces are emitted during tests.
+            // This facilitates debugging when tests fail.
+
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (cgillum) [inline test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs:46]: Can we add comments here with reminders to update these dependencies?

File: test/DurableTask.Test.Orchestrations/SimpleOrchestrations.cs (hunk 11 above)
Comment (cgillum) [reply test/DurableTask.Test.Orchestrations/SimpleOrchestrations.cs:416]: This is the right thing to do given the timeline we're working with, but I'm curious to better understand why this particular pattern isn't working reliably.

File: eng/ci/public-build.yml (hunk 12 above)
Comment (cgillum) [reply eng/ci/public-build.yml:71]: Is the `netcoreapp3.1` part of the path necessary? I see you didn't include it in line 90 and 109, further below.

**Review by cgillum — APPROVED** [review]
Overall LGTM. Left a few comments.

I don't know how to explain the need for explicitly adding System.Runtime.CompilerServices.Unsafe, unless perhaps it's related to differences in how binplacing works on these machines. When running locally, do you see this assembly anywhere in your build output on the file system?

File: test/DurableTask.Test.Orchestrations/SimpleOrchestrations.cs (hunk 11 above)
Comment (davidmrdavid) [reply test/DurableTask.Test.Orchestrations/SimpleOrchestrations.cs:416]: I _think_ I know the root cause - if the ExternalEvent arrives before the ExecutionStarted event, then the `IsExecutableInstance` check will return `false`. We do that here:

https://github.com/Azure/durabletask/blob/d4b98d46158e470c6b769601cd76a86d9bb6d096/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L1038-L1060

So we return "no such instance" as seen above, and we end up **discarding** the message as it is for an instanceID that never existed. That code is here:

https://github.com/Azure/durabletask/blob/d4b98d46158e470c6b769601cd76a86d9bb6d096/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L782-L836

To be clear, the piece where we end up sending external events to the discard list is this particular section:

https://github.com/Azure/durabletask/blob/d4b98d46158e470c6b769601cd76a86d9bb6d096/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L801-L810

So it's an ordering issue. In theory, I think we could be more conservative and not discard events immediately, but that can creating poison messages. Please let me know if I'm missing something.

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs (hunk 13 above)
Comment (davidmrdavid) [reply test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs:47]: Yes, just did here -> https://github.com/Azure/durabletask/pull/1139/commits/bc9a5eab040d4950669c51dfbdd539f0ff568e18

File: test/DurableTask.AzureStorage.Tests/AzureStorageScenarioTests.cs (hunk 3 above)
Comment (bachuv) [reply test/DurableTask.AzureStorage.Tests/AzureStorageScenarioTests.cs:2255]: Yes, I'll add this task to the list.

File: eng/templates/test.yml
Comment (bachuv) [inline eng/templates/test.yml:1]: Can you explain the changes that were made in this file?

File: src/DurableTask.Core/DurableTask.Core.csproj [hunk 14]
```diff
@@ -39,6 +39,7 @@
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="2.2.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="5.0.1" />
+    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="6.0.0" />
```
Comment (jviau) [inline src/DurableTask.Core/DurableTask.Core.csproj:42]: If we are going v3 with DTFx, I think we should try and just upgrade some packages here:

`Microsoft.Extensions.Logging.Abstractions` -> 6.0.0
`System.Diagnostics.DiagnosticSource` -> 6.0.0

See if that fixes your problem

File: test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj (hunk 5 above)
Comment (jviau) [reply test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj:22]: 2.2.0 is quite old. Can we standardize to `6.0.0` as part of the v3 change?

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs [hunk 15]
```diff
@@ -36,23 +38,30 @@ public class DispatcherMiddlewareTests
...
                 .AddTaskActivities(typeof(SimplestGetUserTask), typeof(SimplestSendGreetingTask))
-                .StartAsync();
+                .StartAsync().RunSynchronously();
 
             this.client = new TaskHubClient(service);
         }
 
-        [TestCleanup]
-        public async Task TestCleanup()
+        [TestCleanup()]
```
Comment (jviau) [inline test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs:61]: nit: undo this change

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs [hunk 16]
```diff
@@ -36,23 +38,30 @@ public class DispatcherMiddlewareTests
...
+        public void CleanupTests()
         {
-            await this.worker!.StopAsync(true);
+            this.worker!.StopAsync(true).RunSynchronously();
```
Comment (jviau) [inline test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs:64]: Where is `RunSynchronously` from? Typically you sync wait for tasks with `.GetAwaiter().GetResult();`

**Review by jviau — COMMENTED** [review]
I believe your assembly load issue is a combo of .netstandard and that we reference `System.Diagnostics.DiagnosticSource/5.0.0`, but then Azure Storage brings in a higher version. That package depends on the unsafe package, so I guess that is not transitively being upleveled?

Either way, I don't know if referencing Unsafe directly is the answer vs taking this chance to upgrade DiagnosticSource package.

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs (hunk 15 above)
Comment (davidmrdavid) [reply test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs]: You caught this PR as I was testing some weird stuff here. It's been reverted.

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs (hunk 16 above)
Comment (davidmrdavid) [reply test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs]: As mentioned - You caught this PR as I was testing some weird stuff here. It's been reverted.

File: src/DurableTask.Core/DurableTask.Core.csproj (hunk 14 above)
Comment (davidmrdavid) [reply src/DurableTask.Core/DurableTask.Core.csproj:42]: Ah, maybe that's it. I'll try to diagnostics source change first, that could be it

Comment (davidmrdavid) [reply src/DurableTask.Core/DurableTask.Core.csproj]: No luck with just upgrading DiagnosticsSource -> https://dev.azure.com/azfunc/public/_build/results?buildId=173059&view=ms.vss-test-web.build-test-results-tab .

Let me try to rev the logging abstractions package as well next

File: eng/templates/test.yml
Comment (davidmrdavid) [reply eng/templates/test.yml:1]: Yes. So previously I had:

```yml
  - task: VSTest@2
    displayName: 'Run tests'
    inputs:
      testAssemblyVer2: |
        ${{ parameters.testAssembly }}
        !**\obj\**
        testFiltercriteria: 'TestCategory!=DisabledInCI'
        vsTestVersion: 17.0
        distributionBatchType: basedOnExecutionTime
        platform: 'any cpu'
        configuration: 'Debug'
        diagnosticsEnabled: True
        collectDumpOn: always
        rerunFailedTests: true
        rerunFailedThreshold: 30
        rerunMaxAttempts: 3
```
As you can see, everything below `testAssemblyVer2` is at the same level indentation. That's an issue, since `testFiltercriteria` and the params below are not an input to `testAssemblyVer2`, instead they're properties that are supposed to be at the same level. So I've indented them back/out so that they're properly parsed by 1ES

File: src/DurableTask.Core/DurableTask.Core.csproj (hunk 14 above)
Comment (davidmrdavid) [reply src/DurableTask.Core/DurableTask.Core.csproj]: Still no luck that failed too -> https://dev.azure.com/azfunc/public/_build/results?buildId=173069&view=ms.vss-test-web.build-test-results-tab

I'll keep the packages on the `6.x` version, and also include the `compilerServices.Unsafe` pkg

File: eng/ci/public-build.yml (hunk 12 above)
Comment (davidmrdavid) [reply eng/ci/public-build.yml]: I had `netcoreapp3.1` simply because there is a subset of tests in `net462` that fail to run on 1ES (again, not locally). It's the tests in `test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs`.

I've now changed the pattern back to `'**\bin\**\DurableTask.Core.Tests.dll'` and simply excluded the tests in `test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs` through and `#if !NET462` exclusion. It's hacky - but otherwise the tests fail with the errors seen here: https://dev.azure.com/azfunc/public/_build/results?buildId=173021&view=ms.vss-test-web.build-test-results-tab

File: test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj (hunk 5 above)
Comment (davidmrdavid) [reply test/DurableTask.AzureStorage.Tests/DurableTask.AzureStorage.Tests.csproj:22]: I had some issues getting the code to build on 1ES with more recent versions. I'm open to upgrading, but since this is the test project, I'd like to push that to be a separate PR.

File: src/DurableTask.Core/TaskActivityDispatcher.cs [hunk 17]
```diff
@@ -23,6 +23,7 @@ namespace DurableTask.Core
     using DurableTask.Core.Logging;
     using DurableTask.Core.Middleware;
     using DurableTask.Core.Tracing;
+    using ActivityStatusCode = Tracing.ActivityStatusCode;
```
Comment (davidmrdavid) [inline src/DurableTask.Core/TaskActivityDispatcher.cs:26]: after upgrading DTFx.Core's `System.Diagnostics.DiagnosticSource` to 6.x, there's some ambiguous definitions here. So I needed to add this alias to disambiguate

File: src/DurableTask.Core/Tracing/TraceHelper.cs [hunk 18]
```diff
@@ -85,7 +85,7 @@ public class TraceHelper
             DateTimeOffset startTime = startEvent.ParentTraceContext.ActivityStartTime ?? default;
 
             Activity? activity = ActivityTraceSource.StartActivity(
-                name: activityName,
+                activityName,
```
Comment (davidmrdavid) [inline src/DurableTask.Core/Tracing/TraceHelper.cs:88]: after upgrading DTFx.Core's `System.Diagnostics.DiagnosticSource` to 6.x, these named parameters create ambiguous method invocations here (there's multiple versions of this method with the same parameters in different orders). Using a positional argument here disambiguates the call.

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs [hunk 19]
```diff
@@ -10,6 +10,7 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 //  ----------------------------------------------------------------------------------
+#if !NET462 // for some reasons these tests are not discoverable on 1ES, leading to the test getting aborted. TODO: Needs investigation
```
Comment (davidmrdavid) [inline test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs:13]: FYI @cgillum, @jviau

File: test/DurableTask.Core.Tests/app.config [hunk 20]
```diff
@@ -6,11 +6,5 @@
     <add key="TaskHubName" value="test" />
   </appSettings>
   <runtime>
-    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
-      <dependentAssembly>
-        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-7.0.0.0" newVersion="7.0.0.0" />
-      </dependentAssembly>
-    </assemblyBinding>
```
Comment (davidmrdavid) [inline test/DurableTask.Core.Tests/app.config:14]: we don't use newtonsoft.json v7.0.0 anymore, so this binding redirect is unecessary

File: src/DurableTask.Core/DurableTask.Core.csproj (hunk 14 above)
Comment (davidmrdavid) [reply src/DurableTask.Core/DurableTask.Core.csproj]: I increased the `System.Diagnostics.DiagnosticSource` pkg to `6.0.1` and that **indeed** fixed the issue :-) .
I did opt against increasing the logging abstractions package just because I want to avoid too many transitive dependency changes that may create issues unifying with the rest of bundles. If we have reason to think this is fine, I'm happy to add that in a follow up PR. For now though, I'd like to proceed as-is to unblock a few PRs.

**Comment by davidmrdavid** [comment]
As mentioned [here](https://github.com/Azure/durabletask/pull/1139#discussion_r1682095306), upgrading `System.Diagnostics.DiagnosticSource` to `6.0.1` did fix the DLL loading issue.

I'm going to take a bit of optimistic liberty here to speed up some blocked PRs and merged on the basis of the previous approval. Goes without saying I'm happy to incorporate any follow-up feedback that may arise after this. I just would like to unblock some of the PRs that are waiting for this CI to be in place.

So merging now, but please reach out for feedback / follow up requests. Thanks!

File: test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs (hunk 19 above)
Comment (cgillum) [reply test/DurableTask.Core.Tests/DispatcherMiddlewareTests.cs:13]: Do you mean the `.net462` flavor of the tests are not discoverable? The .NET Core flavor _is_ working?

**Comment by cgillum** [comment]
The latest changes look good to me. ✅

I had one clarifying question about some tests that were disabled, but it's non-blocking.
